### PR TITLE
Dkuhn multiattach tutorial

### PR DESCRIPTION
**Pull request by d-kuhn**

### CODE REVIEW

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+taxonomy:
+    category:
+        - docs
+---
+
+## Cinder multi-attach
+
+### Overview
+
+In this tutorial we will create a cinder volume and attach it to 3 different instances simoultaniously in Read/Write mode using the OCFS2 filesystem.
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:14]: simoultaniously -> simultaneously; file system (2 words)

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+### Prerequisites
+
+* 3 running instances using Ubuntu sharing one network
+* You know the basics of using the [OpenStack CLI-Tools](../../03.Howtos/02.openstack-cli/docs.en.md).
+* Environment variables are set, like shown in the [API-Access-Tutorial](../02.api-access/docs.en.md).
+
+### Create a cinder multi-attach volume
+
+Multiattach volumes are created by choosing the type `quobyte-multiattach`.
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:24]: We should be consistent how to write multi-attach (vs multiattach), so better "Multi-attach volumes" here

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+openstack --os-compute-api-version 2.60 server add volume "App Instance 1" testvolume
+openstack --os-compute-api-version 2.60 server add volume "App Instance 2" testvolume
+openstack --os-compute-api-version 2.60 server add volume "App Instance 3" testvolume
+
+```
+
+If you get an error message like
+>versions supported by client: 2.1 - 2.41
+
+you need to update your openstack client. Working with multi attach volumes is supported from nova API version 2.60.
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:76]: multi-attach -> multi-attach with dash

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+
+### Setup OCFS2
+
+Multi-attach volumes require a [cluster filesystem](https://en.wikipedia.org/wiki/Clustered_file_system#SHARED-DISK) like OCFS2 or GFS2 to coordinate concurrent filesystem access.
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:81]: file system (2 words)

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+**Warning: We do not recommend to use a file system like Ext4 or XFS for multi-attach volumes. Concurrent write access will destroy the filesystem and the data on that volume.**
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:83]: filesystem -> file system

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+For this tutorial we use OCFS2.
+To setup OCFS2 we need to install the ocfs2-tools and kernel module for the running kernel version on every VM:
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:86]: ...and kernel modules...

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+```shell
+apt install ocfs2-tools linux-modules-extra-4.15.0-91-generic
+```
+
+OCFS2 is configurend in /etc/default/o2cb and /etc/ocfs/cluster.conf.
+Set `O2CB_ENABLED=true` in /etc/default/o2cb. All other settings can left unchanged.
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:93]: ...can be left unchanged. ("be" was missing)

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        ip_port = 7777
+        ip_address = 192.168.1.6
+        number = 3
+        name = app-instance-3
+        cluster = ocfs2
+
+cluster:
+    node_count = 3
+    name = ocfs2
+
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:129]: maybe delete the final empty line

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+Start the o2cb service on all 3 VMs and check the status:
+
+```shell
+systemctl start o2cb
+systemctl status o2cb
+systemctl enable o2cb
+```
+
+### Create a filesystem on the volume
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:140]: filesystem -> file system

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+
+In this example there is only one cinder volume attached to the VMs. The device name on all VMs is `/dev/vdb`.
+The next step os only necessary on one VM:
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:143]: typo: The next step os -> is

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+
+```shell
+mkfs.ocfs2 /dev/vdb
+```
+
+### Mount the filesystem
+
+Mount the filesystem on all 3 VMs:
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:151]: filesystem -> file system

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+
+```shell
+mount.ocfs2 /dev/vdb /mnt
+```
+
+Now it is possible to read and write files from all VMs to the volume
```
Comment (bpetermannS11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:157]: Missing period at the end. And "...to the volume" doesn't sound right. Maybe "Now it is possible to read and write files on the volume from all VMs."

Comment (d-kuhn) [reply pages/02.Tutorials/10.cinder-multiattach/docs.en.md]: yes, that sounds better

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md (hunk 6 above)
Comment (d-kuhn) [reply pages/02.Tutorials/10.cinder-multiattach/docs.en.md]: fixed

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md (hunk 7 above)
Comment (d-kuhn) [reply pages/02.Tutorials/10.cinder-multiattach/docs.en.md]: fixed

**Comment by stefreak** [comment]
Cherry-picked the tutorial into #69

File: pages/02.Tutorials/10.cinder-multiattach/docs.en.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+
+## Cinder multi-attach
+
+### Overview
+
+In this tutorial we will create a cinder volume and attach it to 3 different instances simultaneously in Read/Write mode using the OCFS2 file system.
+
+### Prerequisites
+
+* 3 running instances using Ubuntu sharing one network
```
Comment (mbernhardtSys11) [inline pages/02.Tutorials/10.cinder-multiattach/docs.en.md:18]: can we add a link to some howto, that explains what exactly must be done?

**Pull request closed without merging**